src/main/scala/beam/router/BeamRouter.scala

```diff
@@ -396,11 +396,13 @@ class BeamRouter(
   }
 
   private def resolveAddressBlocking(addr: Address, d: FiniteDuration = 60.seconds): Option[ActorRef] = {
-    Try(Await.result(resolveAddress(addr, d), d)).recover {
-      case t: Throwable =>
-        log.error(t, "resolveAddressBlocking failed to resolve '{}' in {}: {}", addr, d, t.getMessage)
-        None
-    }.get
+    Try(Await.result(resolveAddress(addr, d), d))
```

Comment:
Should this one even be wrapped in a Try? It already has a recover, and any errors that come out of this are likely bad enough that they should propagate out.

Reply:
Yes, We dont need Try to Wrap this because its already has a recover